src/client/tile.cpp

```diff
@@ -223,6 +223,88 @@ class SourceImageCache
 			}
 		}
 
+		/* Apply the "clean transparent" filter to textures, removing borders on transparent textures.
+		 * PNG optimizers discard RGB values of fully-transparent pixels, but filters may expose the
+		 * replacement colors at borders by blending to them; this filter compensates for that by
+		 * filling in those RGB values from nearby pixels.
+		 */
+		if (g_settings->exists("texture_clean_transparent")
```

Comment:
Please add a default value for "texture_clean_transparent" in defaultsettings.cpp rather than using this method